compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -583,6 +585,38 @@ generateSrcFromLf env = noLoc mod
             | otherwise
             = pure emptyBag
 
+    interfaceDecls :: [Gen (LHsDecl GhcPs)]
+    interfaceDecls = do
+        iface <- NM.toList $ LF.moduleInterfaces $ envMod env
+        [interfaceName] <- [LF.unTypeConName $ LF.intName iface]
```

Comment:
I noticed that we often explicitly match on a single-component (sometimes double-?) `TypeConName` elsewhere in this file. When do we consider that safe? I guess it makes sense here since we're writing the dummy interface for the module itself, so the type shouldn't be qualified

Reply:
Multi-component names only originate from sums of products, i.e., ADTs of the form
```
data X = A { f : Int } | B { g : Int }
```
which compiles to
```
variant @serializable X =
  | A Test:X.A
  | B Test:X.B
record @serializable X.A = {f : Int64}
record @serializable X.B = {g : Int64}
```
In this particular case, interfaces are never user defined types so there is no way they can be a SoP.

Reply:
aha! got it, thanks. Would it make sense then to change the type of `intName` to a newtype over `Text` instead of `[Text]`?

Reply:
LF does allow multi-component names here so I’d say no. But data-dependencies can assume that code was generated by the compiler and otherwise do stupid stuff/ignore things so here we can be more strict.